boards/Kconfig

```diff
@@ -9,6 +9,14 @@ config BOARD
 	  if not found we look for the linker file in
 	  soc/<arch>/<family>/<series>
 
+config BOARD_REVISION
+	def_string "$BOARD_REVISION"
```

Comment:
@mbolivar-nordic a little late to the game.

I'm sure this is useful but did you notice the fact that the `BOARD` Kconfig setting is not passed through the environment, but set in the board's Kconfig.

The result of that is that the BOARD value in Kconfig != BOARD value passed to CMake, se for example:
https://github.com/zephyrproject-rtos/zephyr/blob/a57dee94d68f2aa0f1a6a07512f1ceab6248c982/boards/arm/nrf5340dk_nrf5340/Kconfig.defconfig#L8-L9

This means that user can fetch the revision for the board given with `-DBOARD=<board>@<rev>`, but the board available in Kconfig may not be the same board, for example:
`-DBOARD=nrf5340dk_nrf5340_cpuapp_ns@1.2.3` results in:
- `CONFIG_BOARD == nrf5340dk_nrf5340_cpuapp`
- `CONFIG_BOARD_REVISION == 1.2.3`

so not sure passing revision this way is wise if we don't cleanup and enforce that `CONFIG_BOARD` is identical to the board provided by the user with `-DBOARD`.

Reply:
> so not sure passing revision this way is wise if we don't cleanup and enforce that `CONFIG_BOARD` is identical to the board provided by the user with `-DBOARD`.

are you aware of any counterexamples? That would indeed seem like a bug in the board defconfig if CONFIG_BOARD ended up different than BOARD (without the revision), wouldn't you agree?

Reply:
> are you aware of any counterexamples? That would indeed seem like a bug in the board defconfig if CONFIG_BOARD ended up different than BOARD (without the revision), wouldn't you agree?

@mbolivar-nordic did you look at the code I linked ?

That is exactly the counter example you are requesting.

A user specifying `-DBOARD=nrf5340dk_nrf5340_cpuapp_ns@0.1.2` will have the BOARD = nrf5340dk_nrf5340_cpuapp_ns is CMake, but as BOARD = nrf5340dk_nrf5340_cpuapp in Kconfig, that is without `_ns` in Kconfig.